cpp/src/groupby/new_groupby.cu

```diff
@@ -219,3 +226,140 @@ gdf_error gdf_group_by(gdf_column* in_key_columns[],
 
   return gdf_error_code;
 }
+
+std::tuple<gdf_index_type*, gdf_size_type> 
+gdf_unique_indices(cudf::table const& input_table, gdf_context* context)
+{
+  gdf_size_type ncols = input_table.num_columns();
+  gdf_size_type nrows = input_table.num_rows();
+
+  rmm::device_vector<void*> d_cols(ncols);
+  rmm::device_vector<int> d_types(ncols, 0);
+  void** d_col_data = d_cols.data().get();
+  int* d_col_types = d_types.data().get();
+
+  bool nulls_are_smallest = (context->flag_null_sort_behavior == GDF_NULL_AS_SMALLEST);
+
+  gdf_index_type* result_end;
+  cudaStream_t stream;
+  cudaStreamCreate(&stream);
+  auto exec = rmm::exec_policy(stream)->on(stream);
+
+  gdf_index_type* unique_indices;
+  RMM_TRY( RMM_ALLOC(&unique_indices, sizeof(gdf_index_type) * nrows, stream) );
+
+  bool const have_nulls{ std::any_of(input_table.begin(), input_table.end(), [](gdf_column const* col){ return col->null_count > 0;}) };
```

Comment:
This was a common enough pattern I made it its own function: https://github.com/rapidsai/cudf/pull/1478/files?file-filters%5B%5D=.hpp#diff-c00c04072fc8516510376c2d9a71bb7cR147

Reply:
done